boa/src/builtins/array/mod.rs

```diff
@@ -1511,6 +1512,180 @@ impl Array {
         new_array.set_field("length", Value::from(new_array_len), true, context)?;
         Ok(new_array)
     }
+    /// `Array.prototype.splice ( start, [deleteCount[, ...items]] )`
+    ///
+    /// Splices an array by following
+    /// The deleteCount elements of the array starting at integer index start are replaced by the elements of items.
+    /// An Array object containing the deleted elements (if any) is returned.
+
+    pub(crate) fn splice(this: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
+        // 1. Let O be ? ToObject(this value).
+        let o = this.to_object(context)?;
+        // 2. Let len be ? LengthOfArrayLike(O).
+        let len = o.length_of_array_like(context)?;
+        // 3. Let relativeStart be ? ToIntegerOrInfinity(start).
+        // 4. If relativeStart is -∞, let actualStart be 0.
+        // 5. Else if relativeStart < 0, let actualStart be max(len + relativeStart, 0).
+        // 6. Else, let actualStart be min(relativeStart, len).
+        let actual_start = Self::get_relative_start(context, args.get(0), len)?;
+        // 7. If start is not present, then
+        let insert_count = if args.get(0).is_none() || args.get(1).is_none() {
+            // 7a. Let insertCount be 0.
+            // 8. Else if deleteCount is not present, then
+            // a. Let insertCount be 0.
+            0
+        // 9. Else,
+        } else {
+            // 9a. Let insertCount be the number of elements in items.
+            args.len().saturating_sub(2)
+        };
+        let actual_delete_count = if args.get(0).is_none() {
+            // 7b. Let actualDeleteCount be 0.
+            0
+            // 8. Else if deleteCount is not present, then
+        } else if args.get(1).is_none() {
+            // 8b. Let actualDeleteCount be len - actualStart.
+            len - actual_start
+        // 9. Else,
+        } else {
+            // b. Let dc be ? ToIntegerOrInfinity(deleteCount).
+            let dc = args.get(1).ok_or(0)?.to_integer_or_infinity(context)?;
```

Comment:
I think this should get the argument or default to undefined, and then use `to_integer_or_infinity()`:
```suggestion
            let dc = args.get(1).cloned().unwrap_or_default().to_integer_or_infinity(context)?;
```